private/buf/cmd/buf/command/plugin/pluginupdate/pluginupdate.go

```diff
@@ -131,6 +133,31 @@ func run(
 	if err != nil {
 		return err
 	}
+	// We do not prune the buf.lock during update, we only add update existing plugin keys
+	// or add new plugin keys.
```

Comment:
This doesn't seem consistent with `buf dep update` which will remove items from the lock file if the dep is removed.

Reply:
Ah yes -- let me revert change.